kolibri/core/content/utils/content_request.py

```diff
@@ -195,6 +197,39 @@ def incomplete_downloads_queryset():
     )
 
 
+def completed_downloads_queryset():
+    """
+    Returns a queryset used to determine the completed downloads, with and without metadata, as
+    well as the total import size if it does have metadata
+    """
+    return (
+        ContentDownloadRequest.objects.filter(status__in=ContentRequestStatus.Completed)
+        .order_by("requested_at")
+        .annotate(
+            has_metadata=Exists(
+                ContentNode.objects.filter(pk=OuterRef("contentnode_id"))
+            ),
+            total_size=_total_size_of_imported_files_annotation(),
+        )
+    )
+
+
+def _total_size_of_imported_files_annotation():
+    """
+    Returns a subquery to determine the total size of imported files
+    """
+    return Subquery(
+        File.objects.filter(
+            Q(contentnode_id=OuterRef("contentnode_id"))
+            | Q(contentnode__parent_id=OuterRef("contentnode_id"))
+            & Q(local_file__available=True)
+        )
+        .annotate(total_size=Sum("local_file__file_size"))
```

Comment:
Just a question out of curiosity here - wondering how much of a performance impact doing this contentnode by contentnode Sum, and then the final Sum of each of these might have, compared to doing the Sum in one go.

Not a blocker, but might be an interesting thing to flag for possible later optimization.

Reply:
After a little more reading, I've found that possibly using the `aggregate()` function within `_total_size_of_imported_files_annotation()` could work to calculate the sum in one go or maybe modifying the `_total_size()` function to use the `sum()` function to return the sum of the results of the `aggregate()` method for each queryset. But I'll defer if that makes sense or actually improves anything to @bjester

Reply:
It seems @LianaHarris360 re-used the existing size annotation (good approach!), which would likely be an expensive operation over many nodes since it uses a subquery for generating the `total_size` field value. Although, I suspect that when used with `_total_size`, and `aggregate` within that, Django's SQL query might get a better plan (in postgres at least) than if the query also selected other fields. So I think its performance is worth taking a looking at, but I suspect it may be okay. It's a dangerous game-- Django ORM and its SQL! :)

Reply:
Yes, I think it would need some empirical testing of the query to be sure - but that's something we can do as a follow up.

Reply:
My existing filter against `total_size` is probably the worst offender. Lets take a look together @LianaHarris360